fe/fe-core/src/test/java/com/starrocks/planner/MaterializedViewTest.java

```diff
@@ -340,11 +340,11 @@ public void testFilter0() {
         String mv = "select empid + 1 as col1 from emps where deptno = 10";
         testRewriteOK(mv, "select empid + 1 from emps where deptno = 10");
         testRewriteOK(mv, "select max(empid + 1) from emps where deptno = 10");
-        testRewriteFail(mv, "select max(empid) from emps where deptno = 10");
+        testRewriteOK(mv, "select max(empid) from emps where deptno = 10").contains("col1 - 1");
 
         testRewriteFail(mv, "select max(empid) from emps where deptno = 11");
         testRewriteFail(mv, "select max(empid) from emps");
-        testRewriteFail(mv, "select empid from emps where deptno = 10");
+        testRewriteOK(mv, "select empid from emps where deptno = 10").contains("col1 - 1");
```

Comment:
have you test it in union case?

Reply:
case about MaterializedViewRewriter#tryUnionRewrite or union in sql ?

Reply:
tryUnionRewrite